### PR TITLE
DBZ-1223 Use LCR position as the record offset

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1223

### CODE REVIEW

**Comment by jpechane** [comment]
Depends on https://github.com/debezium/debezium/pull/846

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/LcrEventHandler.java [hunk 1]
```diff
@@ -50,15 +47,25 @@ public LcrEventHandler(ErrorHandler errorHandler, EventDispatcher<TableId> dispa
 
     @Override
     public void processLCR(LCR lcr) throws StreamsException {
-        long scn = convertPositionToScn(lcr.getPosition());
-
-        // After a restart it may happen we get the event with the last processed SCN again
-        if (scn <= offsetContext.getScn()) {
-            LOGGER.debug("Ignoring change event with already processed SCN {}", scn);
+        LOGGER.trace("Arrived LCR {}", lcr);
```
Comment (gunnarmorling) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/LcrEventHandler.java]: s/Arrived/Received/

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/LcrPosition.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        this.rawPosition = rawPosition;
+        this.stringFromat = Strings.byteArrayToHexString(rawPosition);
+        try {
+            scn = XStreamUtility.getSCNFromPosition(rawPosition).longValue();
+        }
+        catch (SQLException | StreamsException e) {
+            throw new RuntimeException(e);
+        }
+        if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace("LCR position {} converted to wit SCN {}", rawPosition, stringFromat, scn);
```
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/LcrPosition.java]: Btw. I don't think the `isTraceEnabled` is really needed, as just plain parameters are passed to `trace()`, i.e. no costly operations when not-tracing.

**Comment by jpechane** [comment]
@gunnarmorling Requested changes done

**Comment by jpechane** [comment]
@gunnarmorling And yes it was designed to be able to process old offsets too

**Comment by gunnarmorling** [comment]
Applied. Thanks a lot, @jpechane!

**Pull request merged**